pkg/frontend/querymiddleware/codec.go

```diff
@@ -48,6 +48,9 @@ const (
 	statusError = "error"
 
 	totalShardsControlHeader = "Sharding-Control"
+
+	// Instant query specific options
+	instantSplitIntervalControlHeader = "Instant-Split-Interval-Control"
```

Comment:
I would define a single header to control every aspect of the instant split (for now only the internal), similarly to how HTTP `Cache-Control` header works.

```suggestion
	instantSplitControlHeader = "Instant-Split-Control"
```